packages/emotion/src/styleUtils/index.js

```diff
@@ -107,26 +115,38 @@ const withStyle = decorator((ComposedComponent, generateStyle) => {
   const WithStyle = forwardRef((props, ref) => {
     const theme = useTheme()
 
-    return (
-      <ComposedComponent
-        ref={ref}
-        makeStyles={makeStyles(
-          theme,
-          ComposedComponent.displayName,
-          generateStyle,
-          {
-            ...ComposedComponent.defaultProps,
-            ...props
-          }
-        )}
-        {...props}
-      />
-    )
+    // calculates style object,
+    // then applies polyfills before returning it
+    const makeStyles = (data) => {
+      const style = calculateStyles(
+        theme,
+        ComposedComponent.displayName,
+        generateStyle,
+        {
+          ...ComposedComponent.defaultProps,
+          ...props
+        }
+      )(data)
+      return polyfillStyles(style)
+    }
+
+    return <ComposedComponent ref={ref} makeStyles={makeStyles} {...props} />
   })
 
   hoistNonReactStatics(WithStyle, ComposedComponent)
   WithStyle.displayName = `WithStyle(${ComposedComponent.displayName})`
 
   return WithStyle
 })
+
+/**
+ * Applies necessary polyfills on the style object
+ * @param {object} styleObject
+ * @returns {object} the modified style object
```

Comment:
This returns sometimes `undefined` (bidirectional.js line 152), if this is expected it should be mentioned here.

Reply:
```js
function bidirectionalPolyfill(styles) {
  const stylesCopy = {}
  //..... rest of code, forEach ....
  return stylesCopy
```
It is an empty object by default, so it always gives back an object. In the forEach loop it just skips properties with no value.

Reply:
oh yeah, I missed that part, it looks good :)